service-base/src/main/java/org/eclipse/hono/service/quarkus/SamplerProducer.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.service.quarkus;
+
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.quarkus.opentelemetry.runtime.tracing.TracerRuntimeConfig;
+
+/**
+ * A producer for an OpenTelemetry Sampler.
+ */
+@ApplicationScoped
+public class SamplerProducer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SamplerProducer.class);
+
+    @Singleton
+    @Produces
+    Sampler sampler(final TracerRuntimeConfig tracerRuntimeConfig) {
+        final TracerRuntimeConfig.SamplerConfig samplerConfig = tracerRuntimeConfig.sampler;
+        if (!tracerRuntimeConfig.suppressNonApplicationUris) {
+            LOG.info("'quarkus.opentelemetry.tracer.suppress-non-application-uris' set to 'false' - will be ignored");
+        }
+        if (!samplerConfig.parentBased) {
+            LOG.info("'quarkus.opentelemetry.tracer.sampler.parent-based' set to 'false' - will be ignored");
+        }
+
+        Sampler sampler = getBaseSampler(samplerConfig.samplerName, samplerConfig.ratio);
+        sampler = Sampler.parentBased(sampler);
+        sampler = new SamplingPrioritySampler(sampler);
+        // Drop all HTTP request spans created by the quarkus OpenTelemetryVertxTracer for now.
+        // Without http-server metrics enabled, these spans will get the HTTP request path as span name, including
+        // variable parts like device-id, making span selection in the Tracing UI unusable.
+        // Dropping all requests spans also makes sure there are no liveness/readiness request spans created.
+        // Suppression of these spans via "quarkus.opentelemetry.tracer.suppress-non-application-uris" doesn't work
+        // if the non-application-root-path is the same as the overall root path.
```

Comment:
This sounds like we will constantly need to adapt this strategy to development in Quarkus.
What is it that we actually need from Quarkus to better support our use case? Can't we contribute a change to Quarkus instead?

Reply:
The changes in this PR are a first step here, replicating the span names as they were before.
The next step should be removing/adapting the above suppression logic, letting HTTP spans be created via quarkus-opentelemetry and removing the Hono `TracingHandler` (from "opentracing-contrib/java-vertx-web").
Some more changes are needed for that - better to be done in another PR.